### DISCUSSION TITLE
Canonical URL/Alternate URL

### DISCUSSION THREAD

**Discussion by ghost** [mecha-cms/mecha#119] [Q&A] [answered]
Selamat sore Saudara Taufik @taufik-nurrohman  yang sangat baik mendedikasikan waktunya untuk menulis kode CMS Mecha.

Saya mau menanyakan dua pertanyaan dalam dua subbab di bawah ini:

### Tautan Eksternal sebagai URL Kanonis/Tautan Internal sebagai URL Alternatif

Situs koran daring kali pertama membuat publikasi sebuah tulisan yang saya tulis. Artinya URL koran daring tersebut menjadi URL kanonis.

Lantas saya mempublikasikan tulisan saya yang dimuat koran daring di atas di blog pribadi saya. Tentu, dengan demikian, saya tidak boleh membuat URL kanonis dengan konten yang sama persis dengan koran daring di atas, karena ini akan menyebabkan duplikat konten antardua media daring.

Saya ingin menanyakan masalah ini dalam bahasa PHP yang belum saya pahami dengan konsep: jika di dalam YAML ada tautan eksternal, maka buatlah URL saya menjadi tautan alternatif dan tautan eksternal tersebut jadikanlah tautan kanonis.

Bagaimana eksekusi yang tepat atas kode ini:

`<link href="<?= $url->clean; ?>" rel="canonical">`

### Subdomain sebagai URL Alternatif

Dalam kasus ini, saya mengasumsikan ada subdomain dengan konten dan struktur URL sama persis yang ada di Mecha, bedanya cuma dalam bahasa konten.

URL Konten dalam Domain Utama

Konten bahasa Inggris di situs utama:

`https://situs-mecha.net/article/foo-bar-title`

Konten bahasa Indonesia di situs subdomain:

`https://id.situs-mecha.net/article/foo-bar-title`

Bagaimana implementasi atas studi kasus di atas, dalam mana domain utama adalah URL kanonis dan subdomain adalah URL alternatif. Apakah dengan menulis tautan manual laman beranda `https://situs-mecha.net/` lantas memanggil slug yang sama dari domain utama dengan pengkodean PHP.

Semoga teman-teman pengembang Mecha CMS, terutama Mas Taufik, berkenan membantu. Atau, setidaknya ada petunjuk agar saya bisa belajar menulis pernyataan di atas dalam kode PHP.

**Comment by taufik-nurrohman** [comment, marked as answer]
Sore pak!

> Jika di dalam YAML ada tautan eksternal, maka buatlah URL saya menjadi tautan alternatif dan tautan eksternal tersebut jadikanlah tautan kanonis.

Tautan eksternal bisa disimpan sebagai properti `link`, kemudian mungkin bisa dibuat kondisional seperti ini:

``` php
<?php if ($page->link): ?>
  <link href="<?= $page->url; ?>" rel="alternate">
  <link href="<?= $page->link; ?>" rel="canonical">
<?php elseif ($page->url): ?>
  <link href="<?= $page->url; ?>" rel="canonical">
<?php else: ?>
  <link href="<?= $url->clean; ?>" rel="canonical">
<?php endif; ?>
```

> Dalam kasus ini, saya mengasumsikan ada subdomain dengan konten dan struktur URL sama persis yang ada di Mecha, bedanya cuma dalam bahasa konten.

Apakah ini berarti ada dua situs web? Dengan `https://situs-mecha.net` sebagai situs web utama?

> Lantas memanggil slug yang sama dari domain utama dengan pengkodean PHP.

Seperti ini?

``` php
<link href="https://id.situs-mecha.net<?= $url->path; ?>" rel="alternate">
```

**Reply by ghost** [reply]
Baik, terima kasih atas bantuannya.

Iya, Mas, untuk kasus yang kedua ada dua situs.

Nah, untuk saya bisa paham `<?= $url->path; ?>`, `<?= $page->url; ?>`, `<?= $page->link; ?>`, ini saya harus pelajari dari mana? Jika ada data dokumentasinya boleh saya ikut baca?

Biar saya tidak banyak tanya dan menyusahkan Mas Taufik secara terlalu.

Oya saya mengikuti artikel-artikel Mas Taufik di Blogger sejak 2017 sampai sekarang migrasi ke Mecha. Dan saya jadi tertarik untuk mempelajarinya.

Terima kasih atas waktu dan pertolongan jawabannya. 😊

**Reply by taufik-nurrohman** [reply]
Nggak apa-apa pak. Fungsi forum ini memang untuk bertanya-tanya. Karena tidak semua bisa dituliskan ke dalam dokumentasi atau artikel, kadang ada juga beberapa hal yang hanya bisa dijawab melalui contoh kasus.

**Referensi:**

 - https://mecha-cms.com/reference/layout
 - https://mecha-cms.com/reference/class/u-r-l/path
 - https://mecha-cms.com/article/page-property
 - https://mecha-cms.com/store/extension/page
